src/QuDiffProblem.jl

```diff
@@ -5,7 +5,7 @@ Linear ODE Problem definition
 """
 struct QuODEFunction{iip,MType}<: DiffEqBase.AbstractODEFunction{iip}
     linmatrix::Array{MType,2}
-    QuODEFunction(linmatrix::Array{MType,2}) where MType = new{true,ComplexF64}(linmatrix)
+    QuODEFunction(linmatrix::Array{MType,2},::Type{PType} = Float32) where {MType,PType} = new{true,Complex{PType}}(linmatrix)
```

Comment:
What is the type of `MType` is it `Complex{PType}`? then you should infer the precision type from it instead of make user to input an extra argument. We would like to follow the DRY (Don't Repeat Yourself) principal for the interface.